lib/CryptoPluginApi/extensionplugins.cpp

```diff
@@ -15,10 +15,642 @@
+ * set to Sailfish::Secrets::Result::InvalidCollectionError.
+ *
+ * If the secret name specified in the identifier of the \a keyTemplate
+ * duplicates an existing key or secret in the same collection identified by
+ * the collection name specified in the identifier of the \a keyTemplate, the
+ * plugin should return a Sailfish::Crypto::Result with the result code set to
+ * Sailfish::Crypto::Failed and the error code set to
+ * Sailfish::Crypto::Result::InvalidKeyIdentifier and the storage error code
+ * set to Sailfish::Secrets::Result::SecretAlreadyExistsError.
+ *
+ * If the plugin does not support storing new keys, it should return a
+ * Sailfish::Crypto::Result with the result code set to
+ * Sailfish::Crypto::Failed and the error code set to
+ * Sailfish::Crypto::Result::UnsupportedOperation.
+ */
+
+/*!
+ * \fn CryptoPlugin::storedKey(const Sailfish::Crypto::Key::Identifier &identifier, Sailfish::Crypto::Key::Components keyComponents, Sailfish::Crypto::Key *key)
+ * \brief Retrieve the key identified by the specified \a identifier limited
+ *        to those components specified in \a keyComponents and write it
+ *        to the out-parameter \a key.
+ *
+ * If the plugin itself is locked, this function should return a
+ * Sailfish::Crypto::Result with the result code set to
+ * Sailfish::Crypto::Result::Failed and the error code set to
+ * Sailfish::Crypto::Result::CryptoPluginIsLockedError.
+ *
+ * The \a customParameters will contain plugin-specific parameters which may
+ * be required by the plugin.  Such parameters must be documented for clients
+ * in the documentation provided with the plugin, and otherwise should be
+ * ignored by plugin implementers.
+ *
+ * If the collection name specified in the identifier of the \a keyTemplate
+ * does not specify a valid collection managed by the plugin, the plugin
+ * should return a Sailfish::Crypto::Result with the result code set to
+ * Sailfish::Crypto::Failed and the error code set to
+ * Sailfish::Crypto::Result::InvalidKeyIdentifier and the storage error code
+ * set to Sailfish::Secrets::Result::InvalidCollectionError.
+ *
+ * If the secret name specified in the \a identifier does not identify an
+ * existing key or secret in the collection specified in the \a identifier,
+ * the plugin should return a Sailfish::Crypto::Result with the result code
+ * set to Sailfish::Crypto::Failed and the error code set to
+ * Sailfish::Crypto::Result::InvalidKeyIdentifier and the storage error code
+ * set to Sailfish::Secrets::Result::InvalidSecretError.
+ *
+ * If the plugin does not support retrieving key data, it should return a
+ * Sailfish::Crypto::Result with the result code set to
+ * Sailfish::Crypto::Failed and the error code set to
+ * Sailfish::Crypto::Result::UnsupportedOperation.
+ */
+
+/*!
+ * \fn CryptoPlugin::storedKeyIdentifiers(const QString &collectionName, QVector<Sailfish::Crypto::Key::Identifier> *identifiers)
```

Comment:
There is no mention of what should be returned if collectionName is empty. I guess all.

Reply:
Empty should be treated as invalid.  Also the errorCode should be StorageError in this case, which I have fixed now also.  Thanks.